### PR TITLE
Measure raw event size

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this commit we measure raw event size and add it as a new field in
the generated documents.

### CODE REVIEW

File: eventdata/elasticlogs-index.json [hunk 1]
```diff
@@ -12,6 +12,9 @@
       "@timestamp": {
         "type": "date"
       },
+      "_raw_event_size": {
+        "type": "short"
```
Comment (dliappis) [inline eventdata/elasticlogs-index.json]: Are we sure `raw_event` won't ever exceed 32767 characters? Looking at the [message sample](https://github.com/elastic/rally-tracks/tree/master/eventdata#example-document) seems unlikely but not sure how large the `request` string can get.

File: eventdata/parameter_sources/randomevent.py [hunk 2]
```diff
@@ -278,7 +280,20 @@ def generate_event(self):
...
+        # reformat the timestamp as this is not worth the overhead.
+        if self.record_raw_event_size:
+            raw_event = '%s - - [%s] "%s %s HTTP/%s" %s %s "%s" "%s"' % (event["clientip"], event["@timestamp"],
+                                                                         event["verb"], event["request"],
+                                                                         event["httpversion"], event["response"],
+                                                                         event["bytes"], event["referrer"],
+                                                                         event["agent"])
+            raw_event_size = len(raw_event)
+        else:
+            raw_event_size = 0
```
Comment (dliappis) [inline eventdata/parameter_sources/randomevent.py]: I was curious to see whether going through if/else in the existing cases (where `record_raw_event_size` isn't used) affects performance; using a quick `cProfile` based microbenchmark I observed a consistent 1 sec improvement moving `raw_event_size=0` before if and removing `else` with 200million iterations. I guess we can classify this as drop in the ocean?

Conversely when `record_raw_event_size is True` and `raw_event_size = 0` is defined before `if`, therefore reset inside if, it was ~1sec slower, with the same amount of iterations compared to this code.

File: eventdata/parameter_sources/randomevent.py [hunk 3]
```diff
@@ -260,6 +260,8 @@ def __init__(self, params):
             for d in params['delete_fields']:
                 self._delete_fields.append(d.split('.'))
 
+        self.record_raw_event_size = params.get("record_raw_event_size", True)
```
Comment (dliappis) [inline eventdata/parameter_sources/randomevent.py]: If we set this to `True` we will impact all existing benchmarks by having an additional field. Should this default to `False`?

File: eventdata/parameter_sources/randomevent.py [hunk 4]
```diff
@@ -278,7 +280,20 @@ def generate_event(self):
...
+                                                                         event["verb"], event["request"],
+                                                                         event["httpversion"], event["response"],
+                                                                         event["bytes"], event["referrer"],
+                                                                         event["agent"])
+            raw_event_size = len(raw_event)
+        else:
+            raw_event_size = 0
+
         line = '{"@timestamp": "%s", ' \
+               '"_raw_event_size":%d, ' \
```
Comment (dliappis) [inline eventdata/parameter_sources/randomevent.py]: Similarly I am skeptical about having this new field (even with size `0`) for all existing benchmarks.
At the same time, I am aware if were to conditionally build the string depending on the flag, code would be messy and likely have a small perf hit.

File: eventdata/elasticlogs-index.json (hunk 1 above)
Comment (danielmitterdorfer) [reply eventdata/elasticlogs-index.json]: In my tests I saw event sizes up to ~ 1000 bytes so `short` should suffice.

**Comment by danielmitterdorfer** [comment]
@dliappis I've addressed your comments and the new field is now completely optional. As a consequence I had to expose it as a track parameter in every bulk operation (thus the large changeset). Can you please have another look?

File: eventdata/parameter_sources/randomevent.py (hunk 2 above)
Comment (danielmitterdorfer) [reply eventdata/parameter_sources/randomevent.py]: Thanks for double-checking. I'd also consider this a drop in the ocean. Also, the new approach works a bit different anyway (we have completely different code paths to create the document now in order to ensure we don't have any complex code on the hot code path).

File: eventdata/operations/indexing.json [hunk 5]
```diff
@@ -10,82 +10,44 @@ rolledover_indices_suffix_separator: used by the `delete_rolledover_index_patter
 {% set p_translog_sync = translog_sync | default('request') | lower %}
 
 {% set p_disk_type = disk_type | default('ssd') | lower %}
+
+{% set comma = joiner() %}
+{% for bulk_size in [50000, 20000, 10000, 5000, 2000, 1000, 500, 250, 125] %}
```
Comment (dliappis) [inline eventdata/operations/indexing.json]: Nice!

**Pull request merged**